spark/src/test/scala/io/delta/tables/DeltaTableSuite.scala

```diff
@@ -234,9 +234,14 @@ class DeltaTableHadoopOptionsSuite extends QueryTest
   private val publicMethods =
   scala.reflect.runtime.universe.typeTag[io.delta.tables.DeltaTable].tpe.decls
     .filter(_.isPublic)
+    .filter(!_.isSynthetic)
     .map(_.name.toString).toSet
 
-  private val ignoreMethods = Seq()
+  private val ignoreMethods = Seq(
```

Comment:
@Kimahriman I think what you are prototyping here can work already, shall I add tests for these 3 methods for you so that we can have the necessary test coverage?

Reply:
You can try, the only tests to really add are to show that the file system options are ignored in all cases. You can see all the tests are checking for things like

```scala
io.delta.tables.DeltaTable.forPath(spark, path, fsOptions)
```

and making sure _those_ `fsOptions` are respected, that's what the whole suite is about. Currently those options are _not_ respected for any cloning operations, which is why they are in the `ignoredMethods` list without tests